algorithms/alg.burn-verifier.tex

```diff
@@ -0,0 +1,29 @@
+\begin{algorithm}[H]
+    \caption{\label{alg.burn-verifier} A contract for verifying burns from the source chain rooted at genesis $\mathcal{G}_s$. This smart contract runs within the target blockchain.}
+    \begin{algorithmic}[1]
+        \Contract{\textsf{burn-verifier}\text{ extends }\textsf{crosschain}$_{k,m,z}$; \textsf{ERC20}}
+            \State\textsf{mapping}(\textsf{address} $\Rightarrow$ \textsf{uint256}) \textsf{balances}
+            \State{$\textsf{initialized} \gets \emph{false}$;
+                   $\textsf{claimed-events} \gets \emptyset$}
+            \Function{\sf initialize}{$\mathcal{G}_s$}
```

Comment:
The genesis block can be hard coded and the initialize function can be dropped here.